test/factories/user.rb

```diff
@@ -13,6 +13,10 @@
     end
     mfa_hashed_recovery_codes { mfa_recovery_codes.map { |code| BCrypt::Password.create(code) } }
 
+    after :create do |user, evaluator|
+      user.confirm_email! if evaluator.email_confirmed != false && evaluator.unconfirmed_email.blank?
+    end
```

Comment:
Do we need this for every test? Can we set the correct state and then specifically for the password/confirmation token tests, we set the state required there?

Reply:
The problem is that creating a user without a token doesn't work. If you pass in nil you always get a token due to user hooks.

This confirmed state is the correct for every other test. In the password tests we must run `forgot_password!` in many of them (something that was previously skipped because the factory state was bad. It used to be that all users were in a sort of mixed email partially confirmed password reset state.

We could shortcut the email confirmed (the standard user state for most other tests), but it would require preventing the "generate token" hook from running on factory created users. This seemed like the cleanest and most canonical solution.